### PR TITLE
feat: Archive Notes

### PR DESCRIPTION
**Pull request by PeeyushPrashant**
- archive note
- restore note
- delete note from archive

Screenshot (87)
Screenshot (88)
Screenshot (89)

### CODE REVIEW

**Review by Rohanmond — APPROVED** [review]
code is looking good to merge

File: src/components/NoteCard/NoteCard.jsx [hunk 1]
```diff
@@ -1,6 +1,71 @@
...
+import { useAuth } from "../../context/auth-context";
+import { useEffect, useState } from "react";
+
+export const NoteCard=({title,body,editHandler,CreatedAt,backgroundColor,id})=>{
+    
+    
+    const {note,noteState,noteDispatch,archiveState,archiveDispatch}=useData();
+    const {token}= useAuth();
+    const archived= archiveState.archiveData.some((ele)=>ele._id===id);
+    console.log(archived);
```
Comment (surajmore98) [inline src/components/NoteCard/NoteCard.jsx:13]: you can remove this console.log

**Review by surajmore98 — APPROVED** [review]

**Review by ramalingaa — APPROVED** [review]
Code looks clean
This can be merged

**Review by vinc3nati — APPROVED** [review]
looks good

**Pull request merged**